src/test/java/com/gotocompany/firehose/sink/common/blobstorage/BlobStorageFactoryTest.java

```diff
@@ -0,0 +1,227 @@
+package com.gotocompany.firehose.sink.common.blobstorage;
+
+import com.gotocompany.firehose.sink.common.blobstorage.oss.ObjectStorageService;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class BlobStorageFactoryTest {
+
+    private String repeat(String str, int times) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < times; i++) {
+            sb.append(str);
+        }
+        return sb.toString();
+    }
+
+    @Test
+    public void shouldCreateOSSStorageWithBasicConfig() {
+        Map<String, String> config = new HashMap<String, String>() {{
+            put("OSS_TYPE", "SOME_TYPE");
+            put("SOME_TYPE_OSS_BUCKET_NAME", "test-bucket");
+            put("SOME_TYPE_OSS_ENDPOINT", "oss-endpoint");
+            put("SOME_TYPE_OSS_ACCESS_KEY_ID", "access-key");
+            put("SOME_TYPE_OSS_ACCESS_KEY_SECRET", "secret-key");
+        }};
+
+        BlobStorage storage = BlobStorageFactory.createObjectStorage(BlobStorageType.OSS, config);
+        assertTrue(storage instanceof ObjectStorageService);
+    }
+
+    @Test
+    public void shouldCreateOSSStorageWithDirectoryPrefix() {
+        Map<String, String> config = new HashMap<String, String>() {{
+            put("OSS_TYPE", "SOME_TYPE");
+            put("SOME_TYPE_OSS_BUCKET_NAME", "test-bucket");
+            put("SOME_TYPE_OSS_ENDPOINT", "oss-endpoint");
+            put("SOME_TYPE_OSS_ACCESS_KEY_ID", "access-key");
+            put("SOME_TYPE_OSS_ACCESS_KEY_SECRET", "secret-key");
+            put("SOME_TYPE_OSS_DIRECTORY_PREFIX", "prefix");
+        }};
+
+        BlobStorage storage = BlobStorageFactory.createObjectStorage(BlobStorageType.OSS, config);
+        assertTrue(storage instanceof ObjectStorageService);
```

Comment:
Do we have a way to assert whether the SOME_TYPE_OSS_DIRECTORY_PREFIX is passed correctly? Seems like the assertion is similar with the above test case and this test is redundant if the assertion is all the same. 

Same questions with all the positive test cases below, should we also assert the configuration or is it enough just asserting the class of the created object?